ledger/acctupdates.go

```diff
@@ -137,7 +137,7 @@ type accountUpdates struct {
 
 	// resources stored the most recent resource state for every
 	// address&resource that appears in deltas.
-	resources map[accountCreatable]modifiedResource
+	resources map[basics.Address]map[basics.CreatableIndex]modifiedResource
```

Comment:
I don't think that the maintenance cost of having this double-maps is worth it. For the lookupAll, I think that iterating over all the resources is good enough, unless we can demonstrate that it's not the case.
( i.e. iterating over 100K map entries shouldn't take that long for a REST API call; given that the next upgrade would be to reduct the number of in-memory rounds from 320, I can't see why this would be needed ).

Reply:
I wasn't sure how much performance this would actually save, but I know we have users who make a lot of calls to algod's AccountInformation REST handler, so I was thinking of them. I'd argue that until we reduce the number of in-memory rounds from 320 it might be worth benchmarking the performance impact, just to double-check this wouldn't result in a significant degradation for these use cases. It was a pretty small LOC delta to add the extra check to see if you need create the extra map when adding a new entry, for what (possibly) could avoid a big degradation, but I'm just guessing without numbers..

Reply:
You're thinking about the AccountInformation REST endpoint as a common used endpoint. While it is true that this is a common use for this endpoint, the `algod` is general is not meant to be part of a scaling solution. If this optimization was affecting just the REST endpoint, I would be all for that. Unfortunately, it is not. It will definitely create a higher pressure on the GC for the "normal" operational case.
I was also contemplating if I should have proposed to have two maps instead. That approach has the same drawbacks as the previous one, which is why I dismissed it.

Reply:
Please correct me if I'm wrong, but keeping the storage as is would still allow us to get all the resources for a specific index using a single for-loop, right ?
i.e.
```golang
for res, mod := range resources {
    if res.addr != addr {
       continue
    }
    ... // do whatever
}
```
vs.
```golang
for idx, mod := range resources[addr] {
    ... // do whatever
}
```

Reply:
You would find that it's very rare to have many resources per address. Most of our addresses have only one asset associated with them.

Reply:
You could get them all with a single for-loop, I was just thinking of all the addresses+assets you might have to skip looking for the one address you're interested in. Having a map per address just to hold one asset does seem like a waste, I suppose you could optimize for a single-asset-per-address by having `map[basics.Address]struct{single modifiedResource; multiple map[basics.CreatableIndex]modifiedResource}` where the map is nil if there is only one resource held.